mne/epochs.py

```diff
@@ -2700,6 +2700,14 @@ def _concatenate_epochs(epochs_list, with_data=True):
         if epochs.baseline != baseline:
             raise ValueError('Baseline must be same for all epochs')
 
+        # compare event_id
+        common_keys = list(set(event_id.keys()).intersection(
```

Comment:
oh, true with event_id not being dict, I forgot about this.
Regarding your last comment - I don't think concatenating epochs with different event_id keys should be forbidden ie. `concatenate_epochs([epochs['condition1'], epochs['condition2'])` should work. 
Unless different event_id keys map to the same value... I may also add a check for this if it is needed.